dcp/api/job_fs.py

```diff
@@ -0,0 +1,234 @@
+"""
+Job Filesystem.
+
+A vfsystem to be used for dcp jobs via tar vfs.
+
+Note: Currently only supported in the Pyodide worktime.
+Note: This logic will be mostly replaced by dcp-client's JobFS when it lands.
+
+Author: Will Pringle <will@distributive.network>
+Date: July 2024
+"""
+import pathlib
+import tarfile
+import os
+import io
+
+import readline
+import atexit
+import rlcompleter
+
+class JobFS:
+    """Job Virtual Filesystem (JobFS)."""
+    def __init__(self):
+        self.vfs = {}
+        self.vfs['/'] = self.vfs
+        self.home = '/home/pyodide'
+        self.mkdir(self.home)
+        self.cwd = self.home
+
+    def add(self, local_src, vfs_dest=None):
+        """Adds local file[s] to the vfs.
+        If dest not specified, add file from root.
```

Comment:
Should be added from current working dir, not home

Reply:
0a46495e06ddee99f35a159b9569b36a04b02be2